app/javascript/components/visualization/ScatterPlot.jsx

```diff
@@ -941,3 +923,70 @@ function getDragMode(isCellSelecting) {
 }
 
 ScatterPlot.getPlotlyTraces = getPlotlyTraces
+
+/** Intersect filtered cells with dataArrays results */
+export function intersect(filteredCells, scatter) {
+  const intersectedData = {}
+  const dataArrays = scatter.data
+  const keys = Object.keys(dataArrays)
+  keys.forEach(key => intersectedData[key] = [])
+  window.dataArrays = dataArrays
+  // array of cell indexes that are being plotted
+  const plotted = []
+  for (let i = 0; i < filteredCells.length; i++) {
+    const filteredCell = filteredCells[i]
+    const allCellsIndex = filteredCell.allCellsIndex
+    plotted.push(allCellsIndex)
+    for (let j = 0; j < keys.length; j++) {
+      const key = keys[j]
+      const dataArray = dataArrays[key]
+      const filteredElement = dataArray[allCellsIndex]
+      intersectedData[keys[j]].push(filteredElement)
+    }
+  }
+  return [intersectedData, plotted]
+}
+
+ScatterPlot.intersect = intersect
+
+/**
+ * Reassign plot data to change cells filtered by cell faceting to --Filtered-- trace (light grey, transparent)
+ * this keeps the shape of the original plot but clearly shows what cells are excluded
+ *
+ * @param plotted {Array} Indices of points that are to be plotted
+ * @param originalData {Object} original scatter data object
+ * @param filteredData {Object} filtered scatter data object from cell faceting
+ * @returns {Object} reorganized scatter data object with new --Filtered-- trace
+ */
+export function reassignFilteredCells(plotted, originalData, filteredData) {
+  const allIndexes = [...Array(originalData['x'].length).keys()]
+  const originalSet = new Set(allIndexes)
+  const plottedSet = new Set(plotted)
+  const reassignedIndices = [...originalSet].filter(i => !plottedSet.has(i))
```

Comment:
A trivial-but-definite performance optimization: `allIndexes` is an array, then gets transformed into a set (`originalSet`), then gets transformed back to an array `[...originalSet]`.  

However, `originalSet` is never actually used as a set.  Transforming the `plotted` array to `plottedSet` makes sense at a glance due to `!plottedSet.has(i)`, but applying the same transformation to `allIndexes` seems unnecessary.

```suggestion
  const plottedSet = new Set(plotted)
  const reassignedIndices = allIndexes.filter(i => !plottedSet.has(i))
```

Reply:
Ah that's a great catch - will do